server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -194,32 +196,12 @@ private void allocateUnassigned() {
             nextShard: for (int i = 0; i < primaryLength; i++) {
                 final var shard = primary[i];
                 final var assignment = desiredBalance.getAssignment(shard.shardId());
-                final boolean[] isThrottled = { false };
+                final var isThrottled = new AtomicBoolean(false);
                 if (assignment != null) {
-                    final Set<String> assignmentNodeIds;
-
-                    final var forcedNodeIds = allocation.deciders().getForcedInitialShardAllocationToNodes(shard, allocation);
-                    if (forcedNodeIds.isEmpty()) {
-                        assignmentNodeIds = assignment.nodeIds();
-                    } else {
-                        assignmentNodeIds = forcedNodeIds.get();
-                        if (logger.isTraceEnabled()) {
-                            logger.trace(
-                                "Shard assignment is ignored as shard [{}] initial allocation forced to {}",
-                                shard.shardId(),
-                                assignment
```

Comment:
This log statement was lying. This parameter supposed to be ` forcedNodeIds.get()`. Fixed and extracted to a method.